### ISSUE TITLE
Async/concurrent cell execution

### ISSUE THREAD

**Issue by davidbrochart** [ipython/ipykernel#696]
With ipykernel it is already possible to `await` at the top-level. [akernel](https://github.com/davidbrochart/akernel) takes this behavior one step further by launching each cell in a task, allowing for concurrent cell execution. This will open up new possibilities, and it can already be a solution to [difficult problems related to ipywidgets](https://github.com/davidbrochart/akernel/issues/2).
A similar attempt was done in https://github.com/ipython/ipython/pull/11347 by @Carreau, although I think the approach is very different.
Do you think it could be interesting/possible to have ipykernel support this kind of behavior?

cc @ilyabo @kylebarron

**Comment by ilyabo** [comment]
Indeed, akernel solves the [issue with our Jupyter widget](https://github.com/ipython/ipykernel/issues/646). It would be great if this functionality was in ipykernel itself. I think it shouldn't be necessary to install custom kernels just to be able to use `await`.

@davidbrochart There should be the Jupyter Widgets dev meeting at 6:30 CET today. I am planning to join in the first half an hour. It would be great if you could join too, so that we discuss this with the core developers. Here's the [zoom link](https://zoom.us/my/jovyan?pwd=c0JZTHlNdS9Sek9vdzR3aTJ4SzFTQT09
) (hope that it's correct).
